src/pwm.rs

```diff
@@ -102,6 +102,51 @@ pub mod blocking {
             P: Into<Self::Time>;
     }
 
+    impl<T: Pwm> Pwm for &mut T {
+        type Error = T::Error;
+
+        type Channel = T::Channel;
+
+        type Time = T::Time;
+
+        type Duty = T::Duty;
+
+        fn disable(&mut self, channel: &Self::Channel) -> Result<(), Self::Error> {
+            (*self).disable(channel)
+        }
+
+        fn enable(&mut self, channel: &Self::Channel) -> Result<(), Self::Error> {
+            (*self).enable(channel)
+        }
+
+        fn get_period(&self) -> Result<Self::Time, Self::Error> {
+            (**self).get_period()
```

Comment:
Why do some require `**` while most require `*`? Is it to avoid recursion?

Maybe the fully qualified syntax is less footgunny? like this: `T::get_period(self)`

Reply:
I think this is a typo?

Reply:
Yes, it is to avoid recursion. Methods that take `&self` in impls for `&mut T` need `**`, all other cases seem to coerce properly with just a single `*`.

I don't think it is much of a footgun because if it is wrong it will simply fail to compile.